### DISCUSSION TITLE
Choose one of pool_contract_address and pool_public_key

### DISCUSSION THREAD

**Discussion by tksalarzai** [Chia-Network/chia-blockchain#7237] [Support]
While creating plot , when i click on 3 dots on PLOT NFT and enter my details about plot , also i entered Farmer Public Key And Pool Public Key copied from KEYS SECTION of gui ? it gives this error kind of "Choose one of pool_contract_address and pool_public_key" and im unable to start plotting

**Comment by SilverYang** [comment]
I have same issue, what's problem, who can help?
     "**Choose one of pool_contract_address and pool_public_key**"

**Comment by SilverYang** [comment]
from the source code, it seems that do not need to input the pool public key and pool contract address in the edit box,
but if do not need, what's the use of these two edit boxes? 

https://github.com/Chia-Network/chia-blockchain/blob/main/chia/plotting/create_plots.py
-------------------------------------------------------------------------------------
    pool_public_key: Optional[G1Element] = None
    pool_contract_puzzle_hash: Optional[bytes32] = None
    if args.pool_public_key is not None:
        if args.pool_contract_address is not None:
            raise RuntimeError("Choose one of pool_contract_address and pool_public_key")
        pool_public_key = G1Element.from_bytes(bytes.fromhex(args.pool_public_key))
    else:
        if args.pool_contract_address is None:
            # If nothing is set, farms to the provided key (or the first key)
            pool_public_key = get_pool_public_key(args.alt_fingerprint)
        else:
            # If the pool contract puzzle hash is set, use that
            pool_contract_puzzle_hash = decode_puzzle_hash(args.pool_contract_address)
-------------------------------------------------------------------------------------

**Reply by Christopher-Reilly** [reply]
Coming back to bump this 3 years later, as with the update of the plotting commands to include new plotters "chia plotters chiapos ..." is giving me the same error when I follow documentation precisely for this plotter.

https://docs.chia.net/plotting-how-to/#cli-plotting

This section of the docs is saying to use all three, the farmer key, the pool key, and the pool contract address.
"chia plotters chiapos -t < temp dir > -d < destination dir > -f < farmer key > -p < pool key > -c < contract address > -k < size > -n < number of plots >"

When following earlier documentation about farming on many machines, it does not say to include the plotNFT contract address:
https://docs.chia.net/farming-on-many-machines/

"chia plots create -f <farmer_key> -p <pool_key>"

When I follow the old documentation, and include the farmer and pool key on a machine with no private keys, but do not include the pool contract nft address, plot creation begins but is unsuccessful.  No data gets written to disk in < temp dir > and CPU remains idle.  I am reading in the new plotters section of the docs that "chia plots create" command is staying for compatibility, but it appears that if the machine you are plotting from does not include private keys (and therefore can not derive keys naturally without specification of the three keys, I presume), and you follow the updated docs, this old error throws and tells you to choose between pool contract address and pool public key.

Any advice from the team would be appreciated!  Is this a potential bug post plotters update?  I am running chia from the chia-docker repo here, using an image I just pulled within the last two days.
https://github.com/Chia-Network/chia-docker

I am running the container as a harvester only and have setup the harvester protocol to communicate with the farmer successfully through the docker containers with ssl certs, etc. per the farming on many machines docs.  I simply just cannot plot on the harvester.   I am also getting the same error when I attempt to plot on my farmer with the updated docs plotting instructions to include the three keys.

Let me know if I can do anything to help debug this if necessary.  :)


EDIT:  Reading now in the docs that the plotNFT contract is only used when pool farming but can be ignored when solo farming?   Is it correct that the plotting machine needs a key generated to make a plot key?

https://docs.chia.net/plot-public-keys/

**Reply by BrandtH22** [reply]
Hey @tksalarzai , as a heads up we can generally provide more timely and thorough support in our discord server (https://discord.gg/chia). There are a number of things to unravel here:

1. Pool key vs contract address:
- only use one of these based on the type of plot you are trying to create.
- use just the pool key (no contract address) if you are only ever going to farm solo (this is not recommended as a replot is needed to join a pool)
- use just the contract address (no pool key) if you want to use plotNFTs and have the option to join pools using the official protocol (note you can still farm solo with plotNFT plots)

2. What other keys are needed and how to generate:
You need to use a farmer public key and one of the pool identifying keys (read above note to decide which). You can copy the farmer and pool identifier from another machine if you do not have the keyset on the plotting machine, to find the farmer public key follow this guide for either GUI or CLI: https://docs.chia.net/troubleshooting/locating-keys/

3. once you have determined which type of plot (and located the corresponding pool identifier) and also have the farmer public key then you can run the command `chia plotters chiapos -t < temp dir > -d < destination dir > -f < farmer key > -p < pool key > -c < contract address > -k < size > -n < number of plots >` (NOTE only use either the pool key OR the contract address, not both)

**Reply by BrandtH22** [reply]
Also, make sure you review which plotter you want to use. ChiaPos is the original plotter and slower than the new bladebit alternative: https://docs.chia.net/plotting-software/

**Comment by BrandtH22** [comment]
No response from OP, please reopen the ticket if this issue persists
